code-studio/src/js/components/BeeCellEditor.jsx

```diff
@@ -0,0 +1,78 @@
+/* global React */
+
+var BeeCell = require('blockly-mooc/src/maze/beeCell');
+
+module.exports = React.createClass({
+  propTypes: {
+    cell: React.PropTypes.object.isRequired,
+    row: React.PropTypes.number.isRequired,
+    col: React.PropTypes.number.isRequired,
+    onUpdate: React.PropTypes.func.isRequired,
+  },
+
+  handleChange: function (event) {
+    var values = this.props.cell.serialize();
+    values[event.target.name] = (event.target.value === 'undefined' || event.target.value === '') ? undefined : parseInt(event.target.value);
+    var newCell = BeeCell.deserialize(values);
+    this.props.onUpdate(newCell);
+    this.props.cell = newCell;
+  },
+
+  render: function () {
+    var values = this.props.cell.serialize();
```

Comment:
Should we just pass the serialized cell in as a prop instead of the cell itself?